pinot-common/src/main/java/org/apache/pinot/common/function/scalar/regexp/RegexpLikeCiConstFunctions.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar.regexp;
+
+import org.apache.pinot.common.utils.regex.Matcher;
+import org.apache.pinot.common.utils.regex.PatternFactory;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+
+/**
+ * Case-insensitive regexp_like implementation for constant pattern.
+ */
+public class RegexpLikeCiConstFunctions {
```

Comment:
Here is the doc for `REGEXP_LIKE`:
https://docs.oracle.com/cd/B14117_01/server.101/b10759/conditions018.htm

It takes an optional third argument as the `match_parameter`. We can follow the same behavior